corgi/core/authentication.py

```diff
@@ -83,3 +84,45 @@ def has_permission(self, request: Any, view: Any) -> bool:
 
         user_roles = rhat_profile.rhat_roles.strip("[]").split(", ")
         return set(user_roles).intersection(set(view.roles_permitted)) != set()
+
+
+# drf's BasePermission seems to use metaclasses in a way mypy doesn't like
+class RedHatUUIDPermission(BasePermission):  # type: ignore[misc]
+    """A permission class that grants access to users specified by rhatUUID.
+    More than one user can be specified in a non-nested list, set, or tuple
+    of strings or UUIDs."""
+
+    ALLOWED_UUID_TYPES = (list, tuple, set, str, uuid.UUID)
+
+    def has_permission(self, request: Any, view: Any) -> bool:
+        if not hasattr(view, "uuids_permitted"):
+            raise ValueError(f"View {view} doesn't define any permitted UUIDs")
+
+        if type(view.uuids_permitted) not in RedHatUUIDPermission.ALLOWED_UUID_TYPES:
+            raise TypeError(
+                f"View {view} specifies UUIDs with unsupported type {type(view.uuids_permitted)}"
+            )
+
+        if type(view.uuids_permitted) in (list, tuple, set):
+            if not all([type(x) in (str, uuid.UUID) for x in view.uuids_permitted]):
```

Comment:
Breaking this into multiple lines and adding a comment makes this a little clearer, assuming I understood this correctly:
```suggestion
        # if uuids_permitted is a collection, iterate over it to check all members are scalar (str or UUID)
        is_collection = isinstance(view.uuids_permitted, collections.abc.Collection)
        if is_collection and not all(isinstance(member, (str, uuid.UUID)) for member in view.uuids_permitted):
```

Checking each view dynamically seems a little odd. Maybe it's better to assert (in unit tests) that any view requiring authentication defines the uuids_permitted attribute, and populates it with the appropriate types of values.

But the current approach is also fine if you prefer this way, or if I missed some library behavior and this is required.